packages/keystone/session.js

```diff
@@ -0,0 +1,136 @@
+const dayjs = require('dayjs')
+const { Store, Cookie } = require('express-session')
+const { has } = require('lodash')
+
+const conf = require('@open-condo/config')
+const { getCookieOptions } = require('@open-condo/keystone/setup.utils')
+
+const defaultCookieOptions = getCookieOptions(conf)
+
+const INFINITE_LIKE_MAX_AGE = 1000 * (Math.pow(2, 31) - 1) // Around 68 years in milliseconds
+
+const FORBIDDEN_SESSION_FIELDS = [
+    'cookie',
+    'keystoneListKey',
+    'keystoneItemId',
+]
+
+/** @param {String | Number} expires */
+function _makeCookie (expires) {
+    let maxAge
+    if (typeof expires === 'number') {
+        maxAge = expires
+    } else if (typeof expires === 'string') {
+        maxAge = dayjs(expires).valueOf() - dayjs().valueOf()
+    } else {
+        maxAge = INFINITE_LIKE_MAX_AGE
+    }
+
+    /** @type {CookieOptions} */
+    const options = {
+        ...defaultCookieOptions,
+        maxAge,
+    }
+
+    return new Cookie(options).toJSON()
+}
+
+/**
+ * @typedef SessionConfig
+ * @property {String} sessionId - key in store
+ * @property {String} userId - user id
+ * @property {String | Number} expires - number for time span, utc date string for static value
+ * @property {Record<string, any>?} additionalFields - any object to store in session
+ */
+
+/**
+ * Create or update session in store
+ * @param {Store} sessionStore
+ * @param {SessionConfig} config
+ *
+ * @example
+ * // Set session manually, override expire date and add additional fields
+ *
+ * const sessionId = getRandomString()
+ * const userId = user.id
+ * const expiresAt = dayjs().add(2, 'month').toISOString()
+ *
+ * await setSession(context.req.sessionStore, {
+ *      sessionId: getRandomString(),
+ *      userId: user.id,
+ *      expires: expiresAt
+ *      additionalFields: {
+ *          rights: ['canManageIntegrations']
+ *      }
+ *  })
+ *
+ *  // context.req.session:
+ *  /*
+ *      // key name: sess:<sessionId>
+ *      // expires: <close to expiresAt - Date.now(), difference in few seconds / minutes>
+ *     {
+ *          "cookie": {
+ *              "originalMaxAge": 1736261386329,
+ *              "expires": "2024-11-25T11:43:42.920Z",
+ *              "secure": false,
+ *              "httpOnly": true,
+ *              "path": "/",
+ *              "sameSite": "Lax"
+ *          },
+ *          "keystoneListKey": "User",
+ *          "keystoneItemId": "uuid"
+ *          "rights": ["canManageIntegrations]
+ *      }
+ *  *\/
+ *
+ *  await destroySession(context.req.sessionStore, sessionId)
+ *  // context.req.session = null
+ */
+function setSession (sessionStore, config) {
```

Comment:
Think about dynamic listKey as it KS does

Reply:
keystone's auth strategy does not have dynamic list key. It's predefined constant. So let's just allow passing needed listKey